src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,196 @@
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation.setShape(this.switchBackground);
+        getChildren().addAll(this.switchBackground, this.switchButton);
+        //Listener
+        this.switchedOn.addListener((observable, oldValue, newValue) -> {
+            boolean tmpIsOn = newValue.booleanValue();
+            this.switchAnimation.setToX(tmpIsOn ? (44 - 18) : 0);
+            this.fillAnimation.setFromValue(tmpIsOn ? Color.LIGHTGRAY : Color.web("#0099cc"));
+            this.fillAnimation.setToValue(tmpIsOn ? Color.web("#0099cc") : Color.LIGHTGRAY);
+            this.switchTransition.play();
+        });
+        //Mouse listener.
+        setOnMouseClicked(event -> this.switchedOn.set(!this.switchedOn.get()));
+    }
+    //
+    //<editor-fold desc="properties" defaultstate="collapsed">
+    /**
+     * returns switchButton.
+     *
+     * @return Circle
+     */
+    public Circle getSwitchButton() {
+        return this.switchButton;
+    }
+    /**
+     * returns switchBackground.
+     *
+     * @return Rectangle
+     */
+    public Rectangle getSwitchBackground() {
+        return this.switchBackground;
+    }
+    /**
+     * returns switchedOn to change the boolean state of the switch.
+     *
+     * @return SimpleBooleanProperty
+     */
+    public SimpleBooleanProperty getSwitchedOn() {
+        return this.switchedOn;
+    }
+    /**
+     * returns switchAnimation which shows the visual transition of the Circle switchButton.
+     *
+     * @return TranslateTransition
+     */
+    public TranslateTransition getSwitchAnimation() {
+        return this.switchAnimation;
+    }
+    /**
+     * returns fillAnimation to show the color change when clicked on.
+     *
+     * @return FillTransition
+     */
+    public FillTransition getFillAnimation() {
+        return this.fillAnimation;
+    }
+    /**
+     * returns switchTransition to maintain a parallel animation for fillAnimation and switchAnimation.
+     *
+     * @return ParallelTransition
+     */
+    public ParallelTransition getSwitchTransition() {
+        return this.switchTransition;
+    }
+    /**
+     * returns  switchedOnProperty.
+     *
+     * @return BooleanProperty
+     */
+    public BooleanProperty getSwitchedOnProperty() {
+        return this.switchedOn;
+    }
+    /**
+     * returns isSwitchedOn to change boolean state to true.
+     *
+     * @return switch value.
+     */
+    public boolean isSwitchedOn() {
+        return this.switchedOn.get();
+    }
+    /**
+     * sets switchedOn to update new value.
+     *
+     * @param switchedOn boolean
+     */
+    public void setSwitchedOn(boolean switchedOn) {
+        this.switchedOn.set(switchedOn);
+    }
+    /**
+     * returns valueProperty.
+     *
+     * @return BooleanProperty
+     */
+    public BooleanProperty valueProperty() {
+        return this.switchedOn;
+    }
```

Comment:
There are in total three methods that return this property. Are you following some convention here?